src/devices/machine/qvt102.cpp

```diff
@@ -0,0 +1,692 @@
+//  MACHINE EMULATION
+//**************************************************************************
+
+WRITE8_MEMBER(qvt102_device::latch_w)
+{
+	// 7------- host to keyboard
+	// -6------ aux print
+	// --5----- ctc cs0
+	// ---4---- dtri
+	// ----3--- blink
+	// -----2-- rv
+	// ------1- eia/cl
+	// -------0 aux en
+
+	m_latch = data;
+
+	// jumper w3 enables this connection
+	if (machine().phase() != machine_phase::INIT)
+	{
+		if (BIT(m_jumper->read(), 2))
+		{
+			fprintf(stderr, "######## rs232_write_dtr %02x\n", BIT(data, 4));
+			m_rs232_conn_dtr_handler(BIT(data, 4));
+		}
+	}
+
+	m_kbdmcu->set_input_line(MCS48_INPUT_IRQ, BIT(data, 7) ? ASSERT_LINE : CLEAR_LINE);
+}
+
+READ8_MEMBER(qvt102_device::kbd_r)
+{
+	// 7------- keyboard to host
+	// -6------ dtr2
+
+	uint8_t data = 0;
+
+	data |= !(m_aux->dsr_r() || m_aux->dcd_r()) << 6;
+	data |= !m_kbd_data << 7;
+
+	return data;
+}
+
+READ8_MEMBER(qvt102_device::ctc_r)
+{
+	return m_ctc->read(BIT(m_latch, 5));
+}
+
+WRITE8_MEMBER(qvt102_device::ctc_w)
+{
+	m_ctc->write(BIT(m_latch, 5), data);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::acia_txd_w)
+{
+	if (BIT(m_latch, 6) == 0)
+	{
+		// print
+		m_aux->write_txd(state);
+	}
+	else
+	{
+		// aux enable
+		if (BIT(m_latch, 0) == 0)
+			m_aux->write_txd(state);
+
+		fprintf(stderr, "rs232_write_txd %02x\n", state);
+		m_rs232_conn_txd_handler(state);
+	}
+}
+
+WRITE_LINE_MEMBER(qvt102_device::acia_rts_w)
+{
+	m_rs232_conn_rts_handler(state);
+
+	// jumper w4 enables rts output to dtr
+	if (machine().phase() != machine_phase::INIT)
+	{
+		if (BIT(m_jumper->read(), 3))
+		{
+			fprintf(stderr, "######## rs232_write_dtr %02x\n", state);
+			m_rs232_conn_dtr_handler(state);
+		}
+	}
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_dcd_w)
+{
+	if (machine().phase() != machine_phase::INIT)
+		if (BIT(m_jumper->read(), 1))
+			m_acia->write_dcd(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_dsr_w)
+{
+	if (machine().phase() != machine_phase::INIT)
+		if (BIT(m_jumper->read(), 0))
+			m_acia->write_dcd(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_ri_w)
+{
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_cts_w)
+{
+	m_acia->write_cts(state);
+}
+
+WRITE_LINE_MEMBER(qvt102_device::rs232_conn_rxd_w)
+{
+	fprintf(stderr, "qvt102_device::rs232_rxd_w %02x\n", state);
+
+	m_acia->write_rxd(state);
+
+	// aux enable
+	if (BIT(m_latch, 0) == 0)
+		m_aux->write_txd(state);
+}
+
+void qvt102_device::device_resolve_objects()
+{
+	m_rs232_conn_dtr_handler.resolve_safe();
+	m_rs232_conn_rts_handler.resolve_safe();
+	m_rs232_conn_txd_handler.resolve_safe();
+}
+
+void qvt102_device::device_start()
+{
+	// register for save states
+	save_item(NAME(m_latch));
+	save_item(NAME(m_kbd_data));
+	save_item(NAME(m_kbd_bus));
+	save_item(NAME(m_kbd_p1));
+	save_item(NAME(m_kbd_p2));
+}
+
+void qvt102_device::device_reset()
+{
+	m_latch = 0;
+	m_kbd_data = 0;
+	m_kbd_bus = m_kbd_p1 = m_kbd_p2 = 0xff;
```

Comment:
Are these actual latches wired to the machine's system reset?  If not, you should only initialise them in `device_start` and allow the CPUs/MCUs to set them after they're reset.